includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -239,6 +239,11 @@ public function get_product_data() {
 
 		/** @var WC_Product_Variable $product */ // phpcs:ignore
 		$product  = $this->get_product();
+
+		$product_price = $this->get_product_price( $product );
+		if ( ! is_numeric( $product_price ) ) {
```

Comment:
@zmaglica I doubt this is the right place to do the check. `get_product_price` will probably throw error if you create subscription product without price anyways.

`should_show_payment_request_button` method seem to be much better place to add the logic: https://github.com/Automattic/woocommerce-payments/blob/ed958c921d8c54e5e2a7d1bced59ea222f845d81/includes/class-wc-payments-payment-request-button-handler.php#L475-L475